BUG: parase_dates column is in dataframe (#31251)

sathyz opened the pull request:
- [ ] closes #31251 
- [ ] add test_missing_column in pandas/tests/io/parser/test_parse_dates.py
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

sathyz committed:
BUG: parase_dates column is in dataframe (#31251)

sathyz committed:
BUG: parase_dates column is in dataframe (#31251)

MomIsBestFriend commented on pandas/tests/io/parser/test_parse_dates.py:
```diff
@@ -1516,3 +1516,14 @@ def test_hypothesis_delimited_date(date_format, dayfirst, delimiter, test_dateti
+
+def test_missing_column(all_parsers):
+    parser = all_parsers
+    content = StringIO("time,val\n" "212.23, 32\n")
```
I believe that the CI will fail because, this string is not concatenated, can ou please concatenate them?

jschendel commented on pandas/tests/io/parser/test_parse_dates.py:
```diff
@@ -1516,3 +1516,14 @@ def test_hypothesis_delimited_date(date_format, dayfirst, delimiter, test_dateti
+
+
+def test_missing_column(all_parsers):
+    parser = all_parsers
```
Can you add the github issue number as a comment here?

jbrockmendel committed:
REF: collect MultiIndex indexing, set methods (#31553)

jbrockmendel committed:
REF: do all convert_tolerance casting inside Index.get_loc (#31425)

jbrockmendel committed:
REF: move _convert_to_indexer to Loc (#31554)

charlesdong1991 committed:
BUG&TST: df.replace fail after converting to new dtype (#31545)

prakhar987 committed:
TST: added test to validate (#31340) (#31483)

WillAyd commented on pandas/io/parsers.py:
```diff
@@ -3278,6 +3278,8 @@ def _isindex(colspec):
                 if isinstance(colspec, int) and colspec not in data_dict:
                     colspec = orig_names[colspec]
+                elif colspec not in orig_names:
+                    raise ValueError(f"'{colspec}' is not in list")
```
```suggestion
                    raise ValueError(f"Missing column provided to ‘parse_dates’: '{colspec}'")
```

Not tied to the wording but I don’t think “is not in list” is super clear to the end user, so could improve a bit

sathyz replied:
I had missed to check CParser, handled it now.

WillAyd reviewed (changes requested):
Can you add a what’s new message for v1.1?

jbrockmendel committed:
CLN: libperiod (#31527)

sohamtiwari3120 committed:
WEB: Changed whatsnew href from v0.25.0 to v1.0.0 (#31567)

charlesdong1991 committed:
CI: Remove warning raising after new matplotlib release (#31573)

alimcmaster1 committed:
BUG: objToJson.c - fix return value (#31482)

jakevdp committed:
Fix docstring for to_markdown buf parameter (#31565)

jbrockmendel committed:
REF: de-duplicate Period freq conversion code (#31577)

jbrockmendel committed:
REF: organize Series indexing tests (#31557)

glemaitre committed:
BUG accept and deprecate negative integer for max_colwidth (#31569)

alimcmaster1 committed:
CLN: Fix ReadMe Badges (#31579)

jbrockmendel committed:
REF: organize DataFrame tests for indexing, by-method (#31558)

jbrockmendel committed:
REF: CategoricalIndex indexing tests (#31559)

WillAyd committed:
unpinned openpyxl (#31525)

mroeschke committed:
BUG: Timedelta components no longer rounded with high precision integers (#31380)

* BUG: Timedelta components no longer rounded with high precision integers

* Simplify NaT.total_seconds

* Also simplify another tz conversion code path

jbrockmendel committed:
implement test_constructors for Timestamp (#31583)

dsaxton committed:
BUG: Fix qcut for nullable integers (#31440)

nrebena committed:
BUG: Preserve key order when using loc on MultiIndex DataFrame  (#28933)

jbrockmendel committed:
REF: define _convert_to_indexer in Loc (#31585)

YagoGG committed:
BUG: to_json not allowing uploads to S3 (#28375) (#31552)

jbrockmendel committed:
CLN: inherit PeriodIndex._box_func (#31512)

jbrockmendel committed:
CLN: remove IndexEngine.set_value (#31510)

WillAyd committed:
Dead JSON Code Cleanup (#31509)

WillAyd committed:
Cleaned up names (#31508)

jbrockmendel committed:
CLN: No need to use libindex.get_value_at (#31506)

topper-123 committed:
CLN: named parameters for GroupBy.(mean|median|var|std) (#31485)

jbrockmendel committed:
PERF/CLN: float-to-int casting (#31409)

jbrockmendel committed:
REF: consolidate DTI/TDI/PI get_value in ExtensionIndex, associated cleanups (#31406)

simonjayhawkins committed:
TST/CLN: remove setup_method from test_dtypes (#31343)

jbrockmendel committed:
BUG: preserve object dtype for Index set ops (#31401)

jbrockmendel committed:
TST: troubleshoot npdev build (#31594)

kandersolar committed:
BUG: fix reindexing with a tz-aware index and method='nearest' (#31511)

MarcoGorelli committed:
BUG: no longer raise user warning when plotting tz aware time series (#31207)

glemaitre committed:
DOC fix *_option() docstring (#31571)

jorisvandenbossche committed:
DOC: add back google analytics with the new doc theme (#31606)

jorisvandenbossche committed:
DOC: combine regressions in section in v1.0.1 whatsnew (#31607)

* DOC: combine regressions in section in v1.0.1 whatsnew

* move more

* label

* remove bug fixes subsections

* fix rst

rushabh-v committed:
Raise error in read_csv when arguments header and prefix both are not None (#31383)

Closes https://github.com/pandas-dev/pandas/issues/27394

jorisvandenbossche committed:
DOC: add redirect for moved json_normalize docstring (#31622)

TomAugspurger committed:
REGR: Fixed setitem with MultiIndex (#31631)

* REGR: Fixed setitem with MultiIndex

Closes https://github.com/pandas-dev/pandas/issues/31449

* 32-bit compat

drewseibert committed:
replacing .format with f-strings (#31628)

* replacing .format with f-strings

* fixing invalid syntax around the variables being subbed in at the f-strings

* ran pytest and black. passed.

jbrockmendel committed:
BUG: Period[us] start_time off by 1 nanosecond (#31475)

abbiepopa committed:
[MRG] f-string updates for issue #29547 (#31556)

jbrockmendel committed:
CLN: Unreachable branch in Loc._getitem_iterable (#31636)

jbrockmendel committed:
PERF: avoid is_bool_indexer check where possible (#31399)

Commit:
DOC: Fix typo in Getting Started docs (#31642)

jbrockmendel committed:
REGR: Categorical with np.str_ categories (#31528)

* REGR: Categorical with np.str_ categories

TomAugspurger committed:
BUG: Fixed IntervalArray[int].shift (#31502)

TomAugspurger committed:
PERF: Cache MultiIndex.levels (#31651)

* PERF: Cache MultiIndex.levels

Closes https://github.com/pandas-dev/pandas/issues/31648

* fixup tests

TomAugspurger committed:
whatsnew (#31665)

(cherry picked from commit 103840e974a4c8548fccab86cabc6a3161bd94e8)

dsaxton committed:
REGR: Fix TypeError in groupby min / max of period column (#31477)

jbrockmendel committed:
BUG: Block.iget not wrapping timedelta64/datetime64 (#31666)

TomAugspurger committed:
REGR: Fixed handling of Categorical in cython ops (#31668)

Fixed by falling back to the wrapped Python version when the cython
version raises NotImplementedError.

Closes https://github.com/pandas-dev/pandas/issues/31450

jbrockmendel committed:
CLN: assorted cleanups in indexes/ (#31674)

topper-123 committed:
CLN: Remove deprecations (#31675)

jbrockmendel committed:
CLN: misc tslibs (#31673)

jbrockmendel committed:
REF: make _convert_scalar_indexer require a scalar (#31676)

jbrockmendel committed:
CLN: MultiIndex.get_value is a hive of scum and villainy (#31662)

leandermaben committed:
Replaced .format with f- strings (#31660)

MarcoGorelli committed:
check first and last points' labels are correct (#31659)

jorisvandenbossche committed:
BUG: avoid specifying default coerce_timestamps in to_parquet (#31652)

thomasjpfan committed:
CLN Replace format in test_repr_info (#31639)

jbrockmendel committed:
REF: call _maybe_cast_indexer upfront, better exception messages (#31638)

jbrockmendel committed:
BUG: Series.xs boxing datetime64 incorrectly (#31630)

fjetter committed:
Add test for gh 31605 (#31621)

jbrockmendel committed:
CLN: convert_list_indexer is always kind=loc (#31599)

jbrockmendel committed:
REF: implement tests/indexes/objects/ (#31597)

jbrockmendel committed:
REF: parametrize indexing tests (#31592)

jbrockmendel committed:
REF: simplify DTI._parse_string_to_bounds (#31519)

jbrockmendel committed:
REF: _convert_scalar_indexer up-front (#31655)

jbrockmendel committed:
REF: move convert_scalar out of cython (#31672)

jorisvandenbossche committed:
REGR: fix non-reduction apply with tz-aware objects (#31614)

jorisvandenbossche committed:
DOC: fixup v1.0.1 whatsnew entries (#31686)

jorisvandenbossche committed:
TST: add test for regression in groupby with empty MultiIndex level (#31690)

paihu committed:
BUG: read_csv used in file like object RawIOBase is not recognize encoding option (#31596)

TomAugspurger committed:
DOC: Update 1.0.1 release notes (#31699)

TomAugspurger committed:
REGR: Fixed AssertionError in groupby (#31616)

* REGR: Fixed AssertionError in groupby

Closes https://github.com/pandas-dev/pandas/issues/31522

simonjayhawkins committed:
TYP: remove type:ignore from pandas/io/common.py (#31700)

datapythonista committed:
WEB: Link from the website to the docs (#30891)

jorisvandenbossche committed:
DOC: fix contributors listing for v1.0.1 (#31704)

rushabh-v committed:
DOC: add plotting backends in visualization.rst (#31066)

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

SaturnFromTitan committed:
added type annotation to JSONDtype.na_value (#31718)

thomasjpfan committed:
DOC Adds newline to dataframe melt (#31712)

alimcmaster1 committed:
DOC: Add 1.0.2 whatsnew (#31723)

simonjayhawkins committed:
TST/CLN: dtype test_construct_from_string (#31727)

alimcmaster1 committed:
Update travis-37.yaml (#31745)

Remove unrequired channel

jbrockmendel committed:
CLN: inconsistent kwarg name (#31721)

drewseibert committed:
F string fixes (#31751)

alimcmaster1 committed:
BUG: Fix to_excel writers handling of cols (#31729)

jbrockmendel committed:
CLN: inline indexing 1-liners (#31625)

jbrockmendel committed:
organize DTI indexing tests (#31753)

jbrockmendel committed:
REF: Index.get_value call self.get_loc instead of self._engine.get_loc (#31713)

jbrockmendel committed:
API/BUG: make .at raise same exceptions as .loc (#31724)

jbrockmendel committed:
REF: Use _maybe_cast_indexer and dont override Categorical.get_loc (#31681)

jbrockmendel committed:
REF: organize MultiIndex indexing tests (#31715)

jbrockmendel committed:
CLN: remove libindex.get_value_at (#31680)

jbrockmendel committed:
CLN: _convert_scalar_indexer only handle "loc" and "getitem" (#31709)

d1618033 committed:
CI: Add --quiet option to isort command (#31733)

ShaharNaveh committed:
TST: Implement external error raised helper function. (#31130)

@gfyoung wrote this function here
https://github.com/pandas-dev/pandas/issues/30999#issuecomment-575426086
so we can ignore error messages without breaking fututre CI tests of
bare pytest raises calls

Co-authored-by: gfyoung <gfyoung17+GitHub@gmail.com>

xref gh-30999

jbrockmendel committed:
CI: fix feather test (#31791)

jbrockmendel committed:
REF: Move Loc-only methods to Loc (#31589)

* REF: Move Loc-only methods to Loc

* rebase fixup

* merge fixup

ArtificialQualia committed:
BUG: Fixed encoding of pd.NA with to_json (#31748)

jorisvandenbossche committed:
ERR: improve error message for invalid indexer (#31769)

sathyz committed:
BUG: add parse_dates check in cparser (#31251)

sathyz committed:
BUG: add parse_dates check in cparser (#31251)

sathyz committed:
remove colspec check in _process_date_conversion

sathyz committed:
Merge remote-tracking branch 'origin/issue-31251' into issue-31251

sathyz commented:
I had made a mistake on trying to resolve conflict. will have to open a new PR.

MomIsBestFriend commented:
My way around this kind of thing is:
```
git fetch upstream
git rebase upstream/master
```

But it's not a good practice because you'll need to force-push it

```
git push origin branch-name --force
```

sathyz commented:
@MomIsBestFriend  I think I messed up again and got the same problem in #31815.

MomIsBestFriend commented:
@sathyz I reccomend you to read [this](https://datapythonista.me/blog/useful-git-commands.html), which is a blog written by @datapythonista
